packages/types/src/snap.ts

```diff
@@ -210,19 +231,21 @@ export interface ThorchainGetAddressRequest {
   params: ThorchainGetAddressParams
 }
 
-export type AvalancheGetAddressResponse = AvalancheAddress
-export type BinanceGetAddressResponse = BinanceAddress
-export type BitcoinCashGetAddressResponse = BitcoinCashAddress
-export type BitcoinGetAddressResponse = BitcoinAddress
-export type CosmosGetAddressResponse = CosmosAddress
-export type DogecoinGetAddressResponse = DogecoinAddress
-export type EthereumGetAddressResponse = EthereumAddress
-export type KavaGetAddressResponse = KavaAddress
-export type LitecoinGetAddressResponse = LitecoinAddress
-export type OsmosisGetAddressResponse = OsmosisAddress
-export type SecretGetAddressResponse = SecretAddress
-export type TerraGetAddressResponse = TerraAddress
-export type ThorchainGetAddressResponse = ThorchainAddress
+export type MaybeRpcResponse<T> = T | JsonRpcError
```

Comment:
This burned us very, very, very badly.

1. Snaps has a [5 requests queue](https://github.com/MetaMask/snaps/blob/5fce399e21d4521ca58fea24cbb1bf30b5ab1b8b/packages/snaps-controllers/src/snaps/RequestQueue.ts#L18) limit for JSON-RPC calls after which subsequent request will fail until the queue is cleared and we could potentiallly burn through that very easily, which we in fact did.

This resulted in things non-deterministically breaking in a silent way, as our type system was assuming a string.

Without the TypeScript land safety, parsing an object as a string in JS is perfectly valid, and we would end up with weird, random bugs. One example of such weirdness was making requests to THOR with `something something [object object]` as a from address, making the request not return a memo and quotes supposedly fail

2. Even not taking the above in consideration, things can fail